(compat) Added support for layer compat validation at the Runtime / DataStore boundary (#24166)
microsoft/FluidFramework: .changeset/vast-paths-begin.md

---
"@fluidframework/datastore": minor
"@fluidframework/test-runtime-utils": minor
---
---

> "section": legacy

Reviewer: ```suggestion
"section": legacy
includeInReleaseNotes: false
```

Since these new APIs are only intended for internal use, I suggest noting the changes in the changelogs but not the release notes. If there is more customer impact that I'm not seeing, then I suggest updating the description to focus on how customers should adapt to these changes or what they should do in response.

---

> Added ILayerCompatDetails property to FluidDataStoreRuntime, MockFluidDataStoreContext and MockFluidDataStoreRuntime

Reviewer: ```suggestion
New ILayerCompatDetails property on FluidDataStoreRuntime, MockFluidDataStoreContext and MockFluidDataStoreRuntime
```

> An optional property called `ILayerCompatDetails` is added to `FluidDataStoreRuntime`. This is queried by
> `FluidDataStoreContext` in the `Runtime` layer to validate that the `Runtime` and `DataStore` layers are compatible.

Reviewer: ```suggestion
A new optional property, `ILayerCompatDetails`, has been added to `FluidDataStoreRuntime`. This property is used by
`FluidDataStoreContext` in the Runtime layer to validate that the Runtime and DataStore layers are compatible.
```

> `ILayerCompatDetails` is also added to `MockFluidDataStoreRuntime` and `MockFluidDataStoreContext` which are used for

Reviewer: ```suggestion
`ILayerCompatDetails` has also been added to `MockFluidDataStoreRuntime` and `MockFluidDataStoreContext` which are used for
```

> testing because these mock classes are exported as legacy alpha.

Reviewer: ```suggestion
testing.
```

> Note that this property is for internal use only.

Reviewer: ```suggestion
Important: this property is intended for use by Fluid Framework code only. No code outside the Fluid Framework should use or depend on this property in any way.
```

Am I right that this new property would ideally be `@internal`, but it can't be because it's a property on an otherwise public interface?

Author: yes, that's correct - It's a property on a legacy alpha class not interface.
